lib/gst/player/gstplayer.c

```diff
@@ -515,12 +511,16 @@ gst_player_get_property (GObject * object, guint prop_id,
       GST_TRACE_OBJECT (self, "Returning is-playing=%d",
           g_value_get_boolean (value));
       break;
-    case PROP_SUBURI:
+    case PROP_SUBURI: {
+      gchar *suburi = NULL;
+
       g_mutex_lock (&self->lock);
-      g_value_set_string (value, self->suburi);
+      g_object_get (G_OBJECT (self->playbin), "current-suburi", &suburi, NULL);
+      g_value_set_string (value, suburi);
       g_mutex_unlock (&self->lock);
-      GST_DEBUG_OBJECT (self, "Returning has-suburi=%d",
-          g_value_get_boolean (value));
+      GST_DEBUG_OBJECT (self, "Returning has-suburi=%d", suburi ? TRUE : FALSE);
+      g_free (suburi);
```

Comment:
I don't like that we use self->uri for the URI property but playbin's current-suburi for the sub URI property. That seems inconsistent and we should make that more unified